experimental/modular_oak_functions/src/demux.rs

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{Service, ServiceType};
+use std::{collections::BTreeMap, sync::Arc};
+
+/// A stream demultiplexer.
+pub struct Demux {
```

Comment:
Currently `Demux` is only used in `IoListener`. Is the plan to use it anywhere else in future?